Tests/UnitTests/Core/TrackFinding/CombinatorialKalmanFilterTests.cpp

```diff
@@ -48,19 +48,33 @@ namespace Test {
 
 struct ExtendedMinimalSourceLink {
   size_t sourceID = 0;
+  const FittableMeasurement<ExtendedMinimalSourceLink>* measurement = nullptr;
 
-  const FittableMeasurement<ExtendedMinimalSourceLink>* meas{nullptr};
-
-  bool operator==(const ExtendedMinimalSourceLink& rhs) const {
-    return meas == rhs.meas;
-  }
+  ExtendedMinimalSourceLink() = default;
+  ExtendedMinimalSourceLink(
+      size_t sid, const FittableMeasurement<ExtendedMinimalSourceLink>* m)
+      : sourceID(sid), measurement(m) {}
 
   const Surface& referenceSurface() const {
-    return *MeasurementHelpers::getSurface(*meas);
+    return *MeasurementHelpers::getSurface(*measurement);
   }
 
-  const FittableMeasurement<ExtendedMinimalSourceLink>& operator*() const {
-    return *meas;
+  friend bool operator==(const ExtendedMinimalSourceLink& lhs,
+                         const ExtendedMinimalSourceLink& rhs) {
+    return (lhs.measurement == rhs.measurement);
+  }
+  friend bool operator!=(const ExtendedMinimalSourceLink& lhs,
+                         const ExtendedMinimalSourceLink& rhs) {
+    return (lhs.measurement != rhs.measurement);
+  }
```

Comment:
Cosmetic change. Also I don't think you need `operator!=`.

Reply:
Also, (you know my opinion on `friend`, this just changes the operator for the sake of changing it, sorry.

Reply:
Switched to non-`friend` implementations. The `operator!=` is there since the source link concept will be updated in a future PR to require a [regular](https://en.cppreference.com/w/cpp/concepts/regular) type.

Reply:
Ok maybe I'm dumb, but doesn't the compile synthesize `operator!=()` if you don't supply it explicitly? And `regular` only requires the equality operator as per the link you provided.

Reply:
Not as far as I know.

Reply:
I think it does: https://godbolt.org/z/nExK7K

Reply:
I think that is a C++20 feature. See e.g. [cppreference](https://en.cppreference.com/w/cpp/language/operators). If you switch your godbolt example to e.g. gcc 8.3 it does not compile.

Reply:
Indeed, just switching the same GCC 10.2 to -std=c++17 breaks this code.

Reply:
Hm, I had thought C++20 only added synthetic overloads based on `<=>`, but it does seem like the inequality operator was also added at this point only.